integrations/kube-agent-updater/pkg/img/cosign.go

```diff
@@ -125,11 +129,11 @@ func NewCosignSingleKeyValidator(pem []byte, name string, keyChain authn.Keychai
 
 // NamedTaggedToDigest resolves the digest of a reference.NamedTagged image and
 // returns a name.Digest image corresponding to the resolved image.
-func NamedTaggedToDigest(image reference.NamedTagged) (name.Digest, error) {
+func NamedTaggedToDigest(image reference.NamedTagged, opts ...ociremote.Option) (name.Digest, error) {
 	ref, err := name.ParseReference(image.String())
 	if err != nil {
 		return name.Digest{}, trace.Wrap(err)
 	}
-	digested, err := ociremote.ResolveDigest(ref)
+	digested, err := ociremote.ResolveDigest(ref, opts...)
```

Comment:
Note for reviewers: the resolution bug was here, we were not passing the registry options containing the resolved image.